### PR TITLE
[DOCS] Adds anomaly detection example with choropleth map

### PR DESCRIPTION
**Pull request by lcawl**
Related to https://github.com/elastic/kibana/pull/99434 and https://github.com/elastic/kibana/pull/98847

This PR adds an example of how to create anomaly detection jobs that generate "anomaly count by location" maps and how to view that type of data in the Data Visualizer and the Anomaly Explorer.

### Preview

https://stack-docs_1679.docs-preview.app.elstc.co/guide/en/machine-learning/master/mapping-anomalies.html

### CODE REVIEW

File: docs/en/stack/ml/anomaly-detection/mapping-anomalies.asciidoc [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+[testenv="platinum"]
+[[mapping-anomalies]]
+= Mapping anomalies by location
+
+If your data includes vector layers that are defined in the
+{kibana-ref}/maps-connect-to-ems.html[{ems} ({ems-init})], your {anomaly-jobs}
+can generate a map of the anomalies by location.
+
+[discrete]
+[mapping-anomalies-prereqs]
```
Comment (szabosteve) [inline docs/en/stack/ml/anomaly-detection/mapping-anomalies.asciidoc:11]:
```suggestion
[[mapping-anomalies-prereqs]]
```

File: docs/en/stack/ml/anomaly-detection/mapping-anomalies.asciidoc [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+If you want to view choropleth maps in **{data-viz}** or {anomaly-job} results,
+you must have fields that contain valid vector layers
+(such as {elastic-maps-service}/#file/world_countries[country codes] or 
+{elastic-maps-service}/#file/usa_zip_codes[postal codes]).
+
+This example uses the sample web logs data set. For more information, see
+{kibana-ref}/get-started.html#gs-get-data-into-kibana[Add the sample data].
+
+[discrete]
+[visualize-vector-layers]
```
Comment (szabosteve) [inline docs/en/stack/ml/anomaly-detection/mapping-anomalies.asciidoc:23]:
```suggestion
[[visualize-vector-layers]]
```

File: docs/en/stack/ml/anomaly-detection/mapping-anomalies.asciidoc [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+
+If you have fields that contain valid vector layers, you can use the
+**{data-viz}** in the **{ml-app}** app to see a choropleth map, in which each 
+area is colored based on its document count. For example:
+
+[role="screenshot"]
+image::images/weblogs-data-visualizer-choropleth.jpg[A screenshot of a field that contains vector layer values in {data-viz}]
+
+[discrete]
+[mapping-anomalies-jobs]
```
Comment (szabosteve) [inline docs/en/stack/ml/anomaly-detection/mapping-anomalies.asciidoc:34]:
```suggestion
[[mapping-anomalies-jobs]]
```

File: docs/en/stack/ml/anomaly-detection/mapping-anomalies.asciidoc [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+<1> Create the {anomaly-job}.
+<2> Create the {dfeed}.
+<3> Open the job.
+<4> Start the {dfeed}. Since the sample data sets often contain timestamps that
+are later than the current date, it is a good idea to specify the appropriate
+end date for the {dfeed}.
+====
+
+[discrete]
+[mapping-anomalies-results]
```
Comment (szabosteve) [inline docs/en/stack/ml/anomaly-detection/mapping-anomalies.asciidoc:109]:
```suggestion
[[mapping-anomalies-results]]
```

File: docs/en/stack/ml/anomaly-detection/mapping-anomalies.asciidoc [hunk 5]
```diff
@@ -0,0 +1,135 @@
...
+selections. It colors each location to reflect the number of anomalies in that
+selected time period. Locations that have few anomalies are indicated in blue;
+locations with many anomalies are red.
+//TBD: Is it possible to view a legend as is the case in the Data Visualizer?
+Thus you can quickly see the locations that are generating the most anomalies.
+If your vector layers define regions, counties, or postal codes, you can zoom
+in for fine details.
+
+[discrete]
+[mapping-anomalies-next]
```
Comment (szabosteve) [inline docs/en/stack/ml/anomaly-detection/mapping-anomalies.asciidoc:131]:
```suggestion
[[mapping-anomalies-next]]
```

**Review by szabosteve — COMMENTED** [review]
I left a few minor suggestions about the section IDs. Also, the [_Anomaly detection examples_](https://www.elastic.co/guide/en/machine-learning/master/anomaly-examples.html) page aggregates all the examples in a bulleted list. Could you please add this page to that list as well?

File: docs/en/stack/ml/anomaly-detection/mapping-anomalies.asciidoc [hunk 6]
```diff
@@ -0,0 +1,135 @@
...
+in {kib} until you follow the prompts to synchronize the necessary saved objects.
+
+[role="screenshot"]
+image::images/weblogs-anomaly-explorer-vectors.jpg[A screenshot of the anomaly count by location in Anomaly Explorer]
+
+The **Anomaly Explorer** contains a map, which is affected by your swim lane
+selections. It colors each location to reflect the number of anomalies in that
+selected time period. Locations that have few anomalies are indicated in blue;
+locations with many anomalies are red.
+//TBD: Is it possible to view a legend as is the case in the Data Visualizer?
```
Comment (alvarezmelissa87) [inline docs/en/stack/ml/anomaly-detection/mapping-anomalies.asciidoc:125]: Yes, the legend is built-in in the map. It needs to be expanded from the map controls to be viewed.

Comment (lcawl) [reply docs/en/stack/ml/anomaly-detection/mapping-anomalies.asciidoc]: Thanks!  Not sure how I missed seeing that expansion arrow! I've refreshed the screenshot with that legend visible.

**Review by alvarezmelissa87 — APPROVED** [review]
LGTM ⚡

**Pull request merged**